index-import.cpp

```diff
@@ -210,45 +195,89 @@ static bool cloneRecords(StringRef recordsDirectory,
 
     std::error_code failed;
     if (status->type() == fs::file_type::directory_file) {
-      if (not fs::exists(outputPath)) {
-        failed = fs::create_directory(outputPath);
+      failed = fs::create_directory(outputPath);
+      if (failed && failed != std::errc::file_exists) {
+        success = false;
+        errs() << "error: " << strerror(errno) << "\n"
+               << "\tcould not create directory: " << outputPath << "\n";
       }
     } else if (status->type() == fs::file_type::regular_file) {
       // Two record files of the same name are guaranteed to have the same
       // contents (because they include a hash of their contents in their
       // file name). If the destination record file already exists, it
       // doesn't need to be cloned or copied.
-      if (not fs::exists(outputPath)) {
-        failed = fs::copy_file(inputPath, outputPath);
+      failed = fs::copy_file(inputPath, outputPath);
+      if (failed && failed != std::errc::file_exists) {
+        success = false;
+        errs() << "error: " << strerror(errno) << "\n"
+               << "\tcould not copy record file from `" << inputPath << "` to `"
+               << outputPath << "`\n";
       }
     }
-
-    if (failed) {
-      success = false;
-      llvm::errs() << "error: " << strerror(errno) << "\n"
-                   << "\tcould not copy record file: " << inputPath << "\n";
-    }
   }
 
   if (dirError) {
     success = false;
-    llvm::errs() << "error: aborted while reading from records directory: "
-                 << dirError.message() << "\n";
+    errs() << "error: aborted while reading from records directory: "
+           << dirError.message() << "\n";
   }
 
   return success;
 }
 
+// Normalize a path by removing /./ or // from it.
+std::string normalizePath(StringRef Path) {
```

Comment:
Can this not use `fs::real_path`?

Reply:
`fs::real_path`` also makes the paths absolute and I _think_ resolves symlinks which is expensive...

Reply:
Good point on the update. Since it only happens twice, I'd lean toward using `real_path` for simplicity.

Reply:
Can't use `real_path` on non-existent directories (e.g. the output dir).. I say we stay with the custom code.

Reply:
Sounds good. Sorry for sending you down a dead end.

Reply:
np